pkg/config/virtio.go

```diff
@@ -551,6 +562,24 @@ func (dev *VirtioBlk) FromOptions(options []option) error {
 	return dev.DiskStorageConfig.FromOptions(unhandledOpts)
 }
 
+func (dev *VirtioBlk) Validate() error {
+	imgPath := dev.ImagePath
+	file, err := os.Open(imgPath)
+	if err != nil {
+		return fmt.Errorf("failed to open file %s: %v", imgPath, err)
+	}
+	defer file.Close()
+	header := make([]byte, 4)
+	_, err = file.Read(header)
+	if err != nil {
+		return fmt.Errorf("failed to read the header of file %s: %v", imgPath, err)
+	}
+	if string(header) == "QFI\xfb" {
```

Comment:
A bit worried of the conversion from bytes to string when we have no guarantee the data will be valid utf-8.
I’d favour `if header == []byte("QFI\xfb")` (not sure about the exact syntax)